src/models/filesystemBackedImpostersRepository.js

```diff
@@ -129,9 +129,16 @@ function create (config, logger) {
         await fsExtra.ensureDir(dir);
     }
 
+    async function ensureFile (filepath) {
+        fsExtra.close(await fsExtra.open(filepath, 'as'));
+    }
+
     async function writeFile (filepath, obj) {
         await ensureDir(filepath);
-        await fsExtra.writeFile(filepath, JSON.stringify(obj, null, 2));
+        await ensureFile(filepath);
```

Comment:
I could not understand why this call is needed from the PR description. Wasn't it supposed to be just the `flag: 'rs+'` that would solve the issue?

Reply:
No it doesn't. That's because the `rs+` flag doesn't create the file, so you need to do this. The `fs.writeFile` method uses the `w` flag by default which automatically creates the file if it doesn't exist.

Check the flags documentation here: https://nodejs.org/api/fs.html#file-system-flags